test/unit/prop/cnf_stream_white.h

```diff
@@ -136,6 +136,8 @@ class CnfStreamWhite : public CxxTest::TestSuite {
     d_nodeManager = NodeManager::fromExprManager(d_exprManager);
     d_scope = new SmtScope(d_smt);
 
+    // must initialize before getting the theory engine
+    d_smt->finalOptionsAreSet();
```

Comment:
Calling `d_smt->finalOptionsAreSet()` should be entirely done by the SMT engine itself. How about we make `d_theoryEngine` of SMT engine private and add a `getTheoryEngine()` that calls `finalOptionsAreSet()`? This is less error prone.